src/overlays/actors/ovl_En_Honotrap/z_en_honotrap.c

```diff
@@ -24,59 +50,442 @@ const ActorInit En_Honotrap_InitVars = {
+    this->actor.velocity.z = (this->targetPos.z - this->actor.posRot.pos.z) * distFrac;
+    this->speedMod = 0.0f;
+
+    this->timer = 160;
+}
+
+void EnHonotrap_FlameMove(EnHonotrap* this, GlobalContext* globalCtx) {
+    s32 pad;
+    Vec3f speed;
+    s32 ready;
+    Player* player;
+    Vec3f shieldNorm;
+    Vec3f tempVel;
+    Vec3f shieldVec;
+
+    Math_StepToF(&this->speedMod, 13.0f, 0.5f);
+    speed.x = fabsf(this->speedMod * this->actor.velocity.x);
+    speed.y = fabsf(this->speedMod * this->actor.velocity.y);
+    speed.z = fabsf(this->speedMod * this->actor.velocity.z);
+    ready = Math_StepToF(&this->actor.posRot.pos.x, this->targetPos.x, speed.x) & 1;
+    ready &= Math_StepToF(&this->actor.posRot.pos.y, this->targetPos.y, speed.y);
+    ready &= Math_StepToF(&this->actor.posRot.pos.z, this->targetPos.z, speed.z);
+    func_8002E4B4(globalCtx, &this->actor, 7.0f, 10.0f, 0.0f, 0x1D);
+
+    if (this->collider.tris.base.atFlags & 4) {
+        player = PLAYER;
+
+        shieldVec.x = -player->shieldMf.zx;
+        shieldVec.y = -player->shieldMf.zy;
+        shieldVec.z = -player->shieldMf.zz;
+        EnHonotrap_GetNormal(&shieldNorm, &shieldVec);
+
+        tempVel = this->actor.velocity;
+        Math3D_Vec3fReflect(&tempVel, &shieldNorm, &this->actor.velocity);
+        this->actor.speedXZ = this->speedMod * 0.5f;
+        this->actor.posRot.rot.y = Math_Atan2S(this->actor.velocity.z, this->actor.velocity.x);
+        EnHonotrap_SetupFlameVanish(this);
+    } else if (this->collider.tris.base.atFlags & 2) {
+        this->actor.velocity.y = this->actor.speedXZ = 0.0f;
+        EnHonotrap_SetupFlameVanish(this);
+    } else if (this->timer <= 0) {
+        EnHonotrap_SetupFlameVanish(this);
+    } else {
+        EnHonotrap_FlameCollisionCheck(this, globalCtx);
+        if (ready) {
+            EnHonotrap_SetupFlameChase(this);
+        }
+    }
+}
+
+void EnHonotrap_SetupFlameChase(EnHonotrap* this) {
+    this->actionFunc = EnHonotrap_FlameChase;
+
+    this->actor.velocity.x = this->actor.velocity.y = this->actor.velocity.z = this->actor.speedXZ = 0.0f;
+    this->actor.posRot.rot.x = this->actor.posRot.rot.y = this->actor.posRot.rot.z = 0;
+
+    this->timer = 100;
+}
+
+void EnHonotrap_FlameChase(EnHonotrap* this, GlobalContext* globalCtx) {
+    s32 pad;
+    Player* player;
+    Vec3s shieldRot;
+
+    Math_ScaledStepToS(&this->actor.posRot.rot.y, this->actor.yawTowardsLink, 0x300);
+    Math_StepToF(&this->actor.speedXZ, 3.0f, 0.1f);
+    this->actor.gravity = (-this->actor.yDistToLink < 10.0f) ? 0.08f : -0.08f;
+    func_8002D868(&this->actor);
+    if (this->actor.velocity.y > 1.0f) {
+        this->actor.velocity.y = 1.0f;
+    }
+    this->actor.velocity.y *= 0.95f;
+    func_8002D7EC(&this->actor);
+    func_8002E4B4(globalCtx, &this->actor, 7.0f, 10.0f, 0.0f, 0x1D);
+    if (this->collider.cyl.base.atFlags & 4) {
+        player = PLAYER;
```

Comment:
might look nice to move the declares for both vars down here, and put the player declare on the same line